execute all

makarenko09 opened the pull request.

Commit:
execute all

Commit:
execute all tasks of lesson two

AleksandrZhukovJava commented on src/Main.java:
```diff
@@ -1,5 +1,165 @@
+        //task 3 lesson 2
+        short changeYear = 1552;
+        short startLeapYear = 1548;
+        if (changeYear >= startLeapYear) {
+            if (changeYear % 4 == 0) {
+                System.out.println("Год " + changeYear + " является високосным");
+            } else if (changeYear % 400 == 0) {
+                System.out.println("Год " + changeYear + " является високосным");
+            } else {
+                System.out.println("Год " + changeYear + " не является високосным");
+            }
+        } else {
+            System.out.println("Год " + changeYear + " не является високосным");
+        }
```
- двойную вложенность надо убрать, можно использовать несколько if поряд но не внутри друг друга, при желании тут все можно реализовать в один if

AleksandrZhukovJava replied:
двойную вложенность надо убрать, можно использовать несколько if поряд но не внутри друг друга, при желании тут все можно реализовать в один if

AleksandrZhukovJava commented on src/Main.java:
```diff
@@ -1,5 +1,165 @@
+        short radiusMax = 100;
+        short deliveryDistance = 95;
+
+        if (deliveryDistance < radiusMax && deliveryDistance > 0) {
+            System.out.print("Потребуется дней: ");
+            if (deliveryDistance < 20) {
+                System.out.print("одни сутки");
+            } else if (deliveryDistance < 60) {
+                System.out.print("двое суток");
+            } else if (deliveryDistance < 100) {
+                System.out.print("трое суток");
+            }
+        }
```
Тоже нужно убрать двойную вложенность, просто добавь еще один else

AleksandrZhukovJava commented on src/Main.java:
```diff
@@ -1,5 +1,165 @@
+
+        Month month = Month.of(monthNumber);
+        switch (month) {
+            case DECEMBER:
+            case JANUARY:
+            case FEBRUARY:
+                System.out.println("Зима");
+                break;
```
можно 
 case DECEMBER, JANUARY, FEBRUARY:
                System.out.println("Зима");
                break;

AleksandrZhukovJava commented on src/Main.java:
```diff
@@ -1,5 +1,165 @@
+        int monthNumber = 6;
+        System.out.println();
+        /*Прописываем условие в котором программа не будет выполняться*/
+        if (monthNumber < 1 || monthNumber > 12) {
+            System.out.println("Номер месяца должен быть от 1 до 12");
+        }
```
Внутри Month уже есть такая проверка, а если хочется что бы не было ошибки  надо весь последующий код завернуть в этот if и перевернуть условие, или весь последующий код написать в else

Commit:
remark Задача 4 урока 2 (определение сроков доставки): Внутри блока if находится несколько других if-else условий для расчета дней доставки в зависимости от расстояния.
remark Задача 3 урока 2 (проверка на високосный год): Вложенный if используется для проверки года на кратность 4 и 400, если год попадает в диапазон после 1548 года.
fix them

Commit:
others fix

AleksandrZhukovJava commented on src/Main.java:
```diff
@@ -1,5 +1,180 @@
+            largest = three;
+        }
+        System.out.println("Largest number: " + largest);
+        // А можно было бы использовать пузырьковую сортировку...
```
не хотелось бы, это On^2, асимптотика точно не в пользу такого подхода

AleksandrZhukovJava commented on src/Main.java:
```diff
@@ -1,5 +1,180 @@
+
+        //task 2 lesson 2
+        short clientDeviceYear = 2015;
+        String warning = clientDeviceYear < 2015 ? "облегченную" : "обычную";
```
название переменной лучше поменять на iosType или типа того

AleksandrZhukovJava commented on src/Main.java:
```diff
@@ -1,5 +1,180 @@
+            if (year < 1548) {
+                throw new IllegalArgumentException("Год не является високосным");
+            }
+            if (year % 4 == 0 || year % 400 == 0) {
```
тут не хватает условия, сейчас 1900 год - високосный, а это не так

AleksandrZhukovJava commented on src/Main.java:
```diff
@@ -1,5 +1,180 @@
+        //task 3 lesson 2
+        short year = 2021;
+        try {
+            if (year < 1548) {
```
это условия можно было бы добавить к следующим в формате 
if (year > 1584 && (...))
и обойтись без исключений

AleksandrZhukovJava commented on src/Main.java:
```diff
@@ -1,5 +1,180 @@
+        short deliveryDistance = 95;
+        try {
+            if (deliveryDistance < radiusMax && deliveryDistance > 0) {
+                System.out.print("Потребуется дней: ");
```
тут наверное имелось ввиду throw new IllegalArgumentException()

AleksandrZhukovJava commented on src/Main.java:
```diff
@@ -1,5 +1,180 @@
+        switch (Oc) {
+            case "iOS":
+                clientOSf = 0;
+                System.out.println("Установите " + warning + " версию приложения для iOS по ссылке");
```
лог тоже можно было вынести из свитча в котором только выбирать переменную iOS или Android, раз уж начали такой подход

AleksandrZhukovJava commented on src/Main.java:
```diff
@@ -1,5 +1,180 @@
+        //task 2 lesson 2
+        short clientDeviceYear = 2015;
+        String warning = clientDeviceYear < 2015 ? "облегченную" : "обычную";
+        byte clientOSf;
```
на основе этой переменной надо выбирать какой тип устройства, у тебя наоборот
